automation/service/workflow.go

```diff
@@ -582,32 +591,32 @@ func (svc *workflow) Exec(ctx context.Context, workflowID uint64, p types.Workfl
 				}
 			}
 
-			return nil, nil
-		}()
+			if !p.Trace {
+				if t == nil {
+					return nil, WorkflowErrUnknownWorkflowStep()
+				} else if !t.Enabled {
+					return nil, WorkflowErrDisabled()
+				}
+			}
```

Comment:
I'm not sure I understand why this bit exists?
Is this correlated to the exec workflow call where the `p.Trace` is set to `false`? If so, I'm not a fan

Reply:
Should we have another parameter to indicate? We can't use the stack (yet).
Perhaps just add some notes why this is here and we'll solve it later